### ISSUE TITLE
Implement additional gemini backend

### ISSUE THREAD

**Issue by andreashappe** [ipa-lab/hackingBuddyGPT#84]
see https://ai.google.dev/gemini-api/docs/models/gemini

**Comment by lloydchang** [comment]
workaround: use a proxy such as https://github.com/zhu327/gemini-openai-proxy

**Comment by lloydchang** [comment]
Regarding https://developers.googleblog.com/en/gemini-is-now-accessible-from-the-openai-library/

Content safety settings are not supported via Gemini's OpenAI chat completion API, as shown in the example at https://www.reddit.com/r/GoogleGeminiAI/comments/1ffb4by/is_there_a_way_to_change_the_safety_settings/

**Comment by lloydchang** [comment]
Consider a workaround, such as a proxy, that turns off the content safety settings by default.

For example:
`gemini-openai-proxy` at https://github.com/zhu327/gemini-openai-proxy/blob/559085101f0ce5e8c98a94fb75fefd6c7a63d26d/pkg/adapter/chat.go#L230-L246
and
`litellm` at https://github.com/BerriAI/litellm/blob/70aa85af1fc01d30934ee71a703b8d3982420d05/litellm/llms/vertex_ai_and_google_ai_studio/vertex_ai_non_gemini.py#L203-L213

The above nuances are better suited for projects that act as proxies, such as `gemini-openai-proxy`, `litellm`, etc.

Adding a Gemini backend to hackingBuddyGPT would require ongoing maintenance to adapt to continuously evolving changes in Gemini's API, which is in beta (`v1beta`) rather than OpenAI's non-beta (`v1`) API.

For instance, maintainers and contributors might spend time trying to understand and justify why Gemini `v1beta` offers an OpenAI chat completion API but does not fully support content safety settings:
* https://www.reddit.com/r/GoogleGeminiAI/comments/1ffb4by/is_there_a_way_to_change_the_safety_settings/
* https://www.reddit.com/r/LocalLLaMA/comments/1gn3zhh/gemini_is_now_accessible_from_the_openai_library/

In conclusion, using proxies is a more sensible approach, given the current state of the Gemini `v1beta` API, as it helps avoid the overhead of adapting to its ongoing changes.

**Comment by lloydchang** [comment]
Relates to Mac target localhost container via gemini openai proxy https://github.com/ipa-lab/hackingBuddyGPT/pull/94
